vaadin-text-field-flow-demo/src/main/java/com/vaadin/flow/component/textfield/demo/TextFieldView.java

```diff
@@ -59,147 +154,343 @@ private void addFocusShortcut() {
+                        return ValidationResult.error("Is not a valid number");
+                    }
+
+                    if (value.length() != 10) {
+                        return ValidationResult
+                                .error("Length must be 10 digits");
+                    }
+
+                    while (intValue > 0) {
+                        sumDigits += intValue % 10;
+                        intValue = intValue / 10;
+                    }
+
+                    if (sumDigits % 10 != 0) {
+                        return ValidationResult.error("ID is not correct");
+                    }
+
+                    return ValidationResult.ok();
+                }).bind(Person::getId, Person::setId);
+        binder.setBean(person);
+        // end-source-example
+        div.setText(
+                "Valid ID: Use a 10 digit number. The sum of digits must be divisible by 10. For example 1111111111.");
+        textField.setId("text-field-synchronous-id");
+        addCard("Validation", "Synchronous custom validator", textField, div);
+    }
+
+    private void themeVariantsTextAlign() {
+        Div div = new Div();
+        // begin-source-example
+        // source-example-heading: Text align
+        TextField leftTextField = new TextField();
+        leftTextField.setValue("left");
+
+        TextField centerTextField = new TextField();
+        centerTextField.setValue("center");
+        centerTextField.getElement().setAttribute("theme", "align-center");
+
+        TextField rightTextField = new TextField();
+        rightTextField.setValue("right");
+        rightTextField.getElement().setAttribute("theme", "align-right");
+        // end-source-example
+
+        leftTextField.setId("text-field-left-id");
+        centerTextField.setId("text-field-center-id");
+        rightTextField.setId("text-field-right-id");
+        div.add(leftTextField, new Text(" "), centerTextField, new Text(" "),
+                rightTextField);
+        addCard("Theme Variants", "Text align", div);
+    }
+
+    private void themeVariantsSmallSize() {
+        // begin-source-example
+        // source-example-heading: Small size
+        TextField textField = new TextField("Label");
+        textField.setPlaceholder("Text field");
+        textField.getElement().setAttribute("theme", "small");
         // end-source-example
+        addCard("Theme Variants", "Small size", textField);
+    }
 
-        textField.setId("disabled-text-field");
-        Div message = new Div();
-        message.setId("disabled-text-field-message");
-        textField.addValueChangeListener(
-                change -> message.setText("Value changed"));
+    private void styling() {
+
+        Div firstDiv = new Div();
+        firstDiv.setText(
+                "To read about styling you can read the related tutorial in");
+        Anchor firstAnchor = new Anchor(
+                "https://vaadin.com/docs/v13/flow/theme/using-component-themes.html",
```

Comment:
This links to Vaadin 13 demos. We can't remember to update this after new releases.